.github/workflows/publish_package.yml

```diff
@@ -0,0 +1,40 @@
+name: Publish to GitHub package registry
+
+on:
+  workflow_dispatch:
+
+jobs:
+  publish_package:
+    name: Run package publish workflow
+    runs-on: ubuntu-latest
+
+    if: github.repository_owner == 'ballerina-platform'
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17.0.7'
+
+      - name: Set Up Ballerina
+        uses: ballerina-platform/setup-ballerina@v1.1.0
+        with:
+          version: latest
+
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+
+      - name: Build without tests
+        env:
+          packageUser: ${{ github.actor }}
+          packagePAT: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew build -x check -x test
+
+      - name: publish package to GitHub package registry
+        env:
+          publishUser: ${{ github.actor }}
+          publishPAT: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew publish
```

Comment:
In this workflow, you can use the Gradle `release` task to release and publish the artifacts. You can check an example [here](https://github.com/ballerina-platform/ballerina-library/blob/803ed204cf711774a2b5e12fde6bb5ceb3c0512b/.github/workflows/release-package-template.yml#L64).